tokio-trace/src/lib.rs

```diff
@@ -146,10 +146,13 @@ macro_rules! meta {
 macro_rules! cached_filter {
     ($meta:expr, $dispatcher:expr) => {{
         use std::sync::atomic::{AtomicUsize, Ordering, ATOMIC_USIZE_INIT};
+        // TODO: maybe we should bundle all this callsite-specific data together
+        // into a struct or something...
+        static FILTERED_BY: AtomicUsize = ATOMIC_USIZE_INIT;
         static FILTERED: AtomicUsize = ATOMIC_USIZE_INIT;
```

Comment:
I think it might be worth making the call-site cache thread local, since the current dispatcher is tracked per thread, and two separate threads hitting the same call-site shouldn't have to invalidate each other. That seems like a potential follow-up PR...